scripts/storybook/webpack.config.js

```diff
@@ -23,7 +23,7 @@ module.exports = (/** @type {webpack.Configuration} */ config) => {
           options: {
             transpileOnly: true,
             experimentalWatchApi: true,
-            configFile: 'tsconfig.json',
+            configFile: path.join(process.cwd(), 'tsconfig.json'),
```

Comment:
With just `tsconfig.json` here, `ts-loader` was resolving it relative to the directory of *the first file loaded* (which might be from a different package), not CWD.